### ISSUE TITLE
remove hardcoded $HOME, ~/

### ISSUE THREAD

**Issue by ghost** [spf13/spf13-vim#562]
I'd like to remove the hardcoded references to the home directory so i can unclutter my $HOME.  Would such a change be accepted in this project or should i fork it?

**Comment by spf13** [comment]
Can you detail more what you were thinking as an alternative?

**Comment by ghost** [comment]
A few things i wanna have :
- allowing to store `~/.vimviews` , `.neocompletecache` (and similiar cachelike things) under `~/.cache`or `%TEMP%`, or whatever is OS relevant
- having customized files like .local, before, and fork together in one directory that isn't `$HOME`
-  storing the uncustomized spf-vim-3 app directory anywhere but `$HOME`
- respecting `$VIMINIT`  during bootstrap.

My personal goal is to end up with something like this:  http://tlvince.com/vim-respect-xdg
but i don't expect everyone to be interested in [xdg](http://standards.freedesktop.org/basedir-spec/basedir-spec-latest.html) style.

**Comment by spf13** [comment]
I'd totally support this, but we would need to provide functions to clear views etc.

**Comment by mwilliammyers** [comment]
I would love to see this implemented and I would even submit a pull request if this is something that you guys would feel other people would have interest in.

**Comment by spf13** [comment]
happy to take a PR

**Comment by mwilliammyers** [comment]
@spf13 awesome! My vision is that if the user desires, they could have $APP_PATH anywhere they want (right now mine is in ~/.dotfiles) and then .vimviews etc., all .vimrc files, .viminfo etc. (any file that has to do with vim at all, besides .vimrc itself) in ~/.vim

Would it be too much to merge if we incorporated this into PR #631?

**Comment by ghost** [comment]
mkwmms, as long as it's possibel to support the fdo xdg spec, then i'll be happy. thus vimviews being able to be stored on .cache/

**Comment by mwilliammyers** [comment]
@jrobeson For sure. I think the user should basically be able to specify where any file/set of files should go. But yeah, the default for that probably should be .cache/ (I would probably end up setting the defaults to the directories listed in [this](https://tlvince.com/vim-respect-xdg) article you mentioned earlier...

Do you think it would be too much if I incorporated these features into your PR #631? I think a lot of this stuff would coincide with the work you did there...

**Comment by ghost** [comment]
I think #631 should be merged as is. There are only two things that might be questionable.
1. inlining the messages in the functions
2. whether the external interface (environment variables) are correct.

**Comment by mwilliammyers** [comment]
Makes sense. I'll wait until its merged to master before submitting a PR... 

My humble opinion: I have been using your branch from #631 for a while now and it has worked great for me. I used all of the environment variables no problem. I can't speak for the windows install though as I have only tested it on Ubuntu, RedHat, and OS X 10.10 The messages seem fine to me... Makes the code more readable.

**Comment by ghost** [comment]
@spf13 : so what do you think about merging #631 then?

**Comment by ghost** [comment]
@mkwmms : i've just been waiting for @spf13's ok for this whole time.

**Comment by spf13** [comment]
@jrobeson I'll merge it today.

**Comment by spf13** [comment]
#631 is now merged.

**Comment by mwilliammyers** [comment]
@spf13 thanks!

**Comment by ghost** [comment]
@mkwmms : are you currently working on this?

**Comment by mwilliammyers** [comment]
@jrobeson sorry, just saw this. I have been busy with another project but I am still planning on implementing XDG over the next few weeks.

**Comment by mwilliammyers** [comment]
@jrobeson Finally started working on this! https://github.com/mkwmms/spf13-vim

I am debating on how far to take XDG support seeing as we might be moving to neovim at some point which will have it by default (so my focus will be making sure spf13-vim specific files support XDG)

**Comment by voxadam** [comment]
Is work, and XDG compliance in general, still being actively pursued by anyone?

**Comment by ghost** [comment]
https://github.com/spf13/spf13-vim/pull/814
